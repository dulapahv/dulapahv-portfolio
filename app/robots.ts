import type { MetadataRoute } from 'next';

import { BASE_URL } from '@/lib/constants';

export default function robots(): MetadataRoute.Robots {
  return {
    rules: [
      {
        userAgent: '*',
        allow: '/'
      }
    ],
    host: BASE_URL
  };
}
