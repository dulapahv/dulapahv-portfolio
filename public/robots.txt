# *
User-agent: *
Allow: /
Disallow: /_next
Disallow: /server-sitemap.xml
Disallow: /404
Disallow: /500
Disallow: /_offline

# Host
Host: https://dulapahv.dev

# Sitemaps
Sitemap: https://dulapahv.dev/sitemap.xml
Sitemap: https://dulapahv.dev/server-sitemap.xml
