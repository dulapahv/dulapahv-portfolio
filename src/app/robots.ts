import { MetadataRoute } from 'next';

import { BASE_URL } from '@/lib/constants';

export default function robots(): MetadataRoute.Robots {
  return {
    rules: [
      {
        userAgent: '*',
        allow: '/',
        disallow: ['/pinku.png', '/ao.png', '/fallback.png'],
      },
      {
        userAgent: 'ia_archiver',
        allow: '/',
      },
    ],
    sitemap: `${BASE_URL}/sitemap.xml`,
    host: BASE_URL,
  };
}
