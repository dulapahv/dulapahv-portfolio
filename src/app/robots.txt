User-agent: *
Allow: /

Sitemap: https://dulapahv.dev/sitemap.xml

Host: https://dulapahv.dev
