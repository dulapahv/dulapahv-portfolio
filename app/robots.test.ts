import { describe, expect, it } from 'vitest';

import robots from './robots';

describe('robots', () => {
  it('should return robots configuration with wildcard user agent', () => {
    const result = robots();

    expect(result.rules).toBeDefined();
    expect(Array.isArray(result.rules)).toBe(true);
    expect(result.rules).toHaveLength(1);

    const rules = Array.isArray(result.rules) ? result.rules : [result.rules];
    const firstRule = rules[0];
    if (firstRule && 'userAgent' in firstRule) {
      expect(firstRule.userAgent).toBe('*');
      expect(firstRule.allow).toBe('/');
    }
  });

  it('should return host URL', () => {
    const result = robots();

    expect(result.host).toBeDefined();
    expect(typeof result.host).toBe('string');
    expect(result.host).toBeTruthy();
  });

  it('should allow all paths for all user agents', () => {
    const result = robots();

    const rules = Array.isArray(result.rules) ? result.rules : [result.rules];
    const firstRule = rules[0];
    if (firstRule && 'allow' in firstRule) {
      expect(firstRule.allow).toBe('/');
    }
  });

  it('should match snapshot', () => {
    const result = robots();
    expect(result).toMatchSnapshot();
  });
});
