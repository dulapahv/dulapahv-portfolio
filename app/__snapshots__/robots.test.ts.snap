// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`robots > should match snapshot 1`] = `
{
  "host": "https://dulapahv.dev",
  "rules": [
    {
      "allow": "/",
      "userAgent": "*",
    },
  ],
}
`;
